source/ui/beamline/AMSlitView.cpp

```diff
@@ -0,0 +1,166 @@
+#include "AMSlitView.h"
+#include "beamline/AMSlit.h"
+#include "ui/beamline/AMExtendedControlEditor.h"
+
+AMSlitView::AMSlitView(AMSlit *slit, bool showBladeEditors, QWidget *parent) :
+	QWidget(parent)
+{
+	// Initialize class variables.
+
+	slit_ = 0;
+	showBladeEditors_ = false;
+
+	// Create UI elements.
+
+	gapEditor_ = new AMExtendedControlEditor(0);
+	gapEditor_->setTitle("Gap");
+
+	centerEditor_ = new AMExtendedControlEditor(0);
+	centerEditor_->setTitle("Center");
+
+	openButton_ = new QPushButton("Open");
+	connect( openButton_, SIGNAL(clicked(bool)), this, SLOT(onOpenButtonClicked()) );
+
+	closeButton_ = new QPushButton("Close");
+	connect( closeButton_, SIGNAL(clicked(bool)), this, SLOT(onCloseButtonClicked()) );
+
+	QHBoxLayout *buttonsLayout = new QHBoxLayout();
+	buttonsLayout->addWidget(openButton_);
+	buttonsLayout->addWidget(closeButton_);
+
+	firstBladeEditor_ = new AMExtendedControlEditor(0);
```

Comment:
Right now, I don't think the editors here have limits on their ranges. The bug that made them necessary before should be taken care of.